src/stan_math_backend/Statement_gen.ml

```diff
@@ -3,9 +3,20 @@ open Middle
 open Fmt
 open Expression_gen
 
+let n_profiles = ref 0
 let pp_call_str ppf (name, args) = pp_call ppf (name, string, args)
 let pp_block ppf (pp_body, body) = pf ppf "{@;<1 2>@[<v>%a@]@,}" pp_body body
 
+let pp_profile ppf (pp_body, name, body) =
+  let profile1 =
+    Fmt.strf
+      "profile<local_scalar_t__> profile_%d__(%s, \
+       const_cast<profile_map&>(profiles__));"
+      !n_profiles name
```

Comment:
I don't like keeping a global `n_profiles` ref _just_ so you can give every profile a unique name. The C++ name of the profile object should not matter and can be reused even for nested profiles--it's always the first declaration in a new scope, and it's allowed to shadow variables in outer scopes.
If you don't want them all to have the same name you could generate the name from statement's location metadata.

Reply:
Ah, of course. That didnt occur to me that its not a problem with nested... Thanks. For the previous language interface we had to be careful because we could have multiple profiles in the same level of nesting. Its not a problem here so `profile__` it is.